### ISSUE TITLE
Input validation: Configuration Wizard

### ISSUE THREAD

**Issue by afercia** [Yoast/yoast-components#843]
See https://github.com/Yoast/wordpress-seo/issues/12173.

All the Configuration Wizard inputs that require a specific format should use an input validation mechanism and provide proper feedback to users. 

For the Social Accounts URL fields, there's a PR exploring browsers constraint validation, see #827.

Other fields that need some input validation mechanism:

Google authorization code:
- it is currently possible to enter any value
- no feedback is given
- it is possible to move to the next step even after entering a clearly invalid value

<img width="717" alt="screenshot 2019-03-06 at 14 02 21" src="https://user-images.githubusercontent.com/1682452/53883517-79015000-4019-11e9-97df-2084c59cf85b.png">

Newsletter email:
- the entered value is validated, however the feedback is insufficient
- the item in error is not clearly identified
- the item in error is not styled
- it is possible to move to the next step even after entering a clearly invalid value
- **important**: the messages come from the Mailchimp API and are not translated

Good news:
- the visible message area is an ARIA live region so error (and success) messages are announced to assistive technologies
- there are proper suggestions for correction 

<img width="400" alt="52333373-61747e80-29fd-11e9-82d1-0a36b89127be" src="https://user-images.githubusercontent.com/1682452/53883832-7521fd80-401a-11e9-950d-9e597245885d.png">
<img width="660" alt="screenshot 2019-03-06 at 14 02 55" src="https://user-images.githubusercontent.com/1682452/53883833-7521fd80-401a-11e9-8007-a1f83d1d8cd5.png">
<img width="382" alt="screenshot 2019-03-06 at 14 15 27" src="https://user-images.githubusercontent.com/1682452/53883835-7521fd80-401a-11e9-8bf0-88767b6aeef6.png">

Not translated messages:

<img width="669" alt="screenshot 2019-03-06 at 14 35 16" src="https://user-images.githubusercontent.com/1682452/53885797-e2379200-401e-11e9-8f69-cd5f0d2b56c0.png">

**Comment by jdevalk** [comment]
I see three different issues here:

* **Google authorization**
  I don't really care about this one for now, for all sorts of reasons, happy to explain on Slack.
* **MailChimp error positioning**
  Seems like we should move that error slightly and make it a bit easier to understand what it relates to, as well as style the input field on error. 
* **MailChimp error translation**
We should see whether we can (automatically and hidden) specify a form language based on the user language in the MailChimp form, as I think MailChimp does have translated error messages available.

**Comment by afercia** [comment]
Closing in favor of https://github.com/Yoast/javascript/issues/137 and https://github.com/Yoast/javascript/issues/138
